node/src/liquidation_worker.rs

```diff
@@ -0,0 +1,582 @@
+								else { return };
+
+							if let Ok(Some(liquidation_option)) = money_market_data
+								.get_best_liquidation_option(&user_data, config.target_hf.into(), (base_asset_address, price.into())) {
+
+								let (Some(collateral_asset_id), Some(debt_asset_id)) =
+									(HydraErc20Mapping::decode_evm_address(liquidation_option.collateral_asset),
+									 HydraErc20Mapping::decode_evm_address(liquidation_option.debt_asset))
+									else {
+										continue
+									};
+
+								let Ok(debt_to_liquidate) = liquidation_option.debt_to_liquidate.try_into()
+									else {
+										continue
+									};
+
+								let liquidation_tx = RuntimeCall::Liquidation(pallet_liquidation::Call::liquidate {
+									collateral_asset: collateral_asset_id,
+									debt_asset: debt_asset_id,
+									user: borrower.0,
+									debt_to_cover: debt_to_liquidate,
+									route: BoundedVec::new(),
+								});
+
+
+								// dry run to prevent spamming with extrinsics that will fail (e.g. because of not being profitable)
+								let dry_run_result = Runtime::dry_run_call(
+									hydradx_runtime::RuntimeOrigin::none().caller,
+									liquidation_tx.clone());
+
+								if let Ok(call_result) = dry_run_result {
+									if call_result.execution_result.is_err() {
+										tracing::debug!(target: LOG_TARGET, "Dry running liquidation failed: {:?}", call_result.execution_result);
+										continue
+									}
+								}
+
+								let encoded_tx: fp_self_contained::UncheckedExtrinsic<hydradx_runtime::Address, RuntimeCall, hydradx_runtime::Signature, hydradx_runtime::SignedExtra> =
+									fp_self_contained::UncheckedExtrinsic::new_unsigned(liquidation_tx);
+								let encoded = encoded_tx.encode();
+								let opaque_tx = sp_runtime::OpaqueExtrinsic::decode(&mut &encoded[..]).expect("Encoded extrinsic is always valid");
+
+								let tx_pool_cc = tx_pool_c.clone();
+								// `tx_pool::submit_one()` returns a Future type, so we need to spawn a new task
+								spawner_c.spawn(
+									"liquidation-worker-on-submit",
+									Some("liquidation-worker"),
+									async move {
+										tracing::debug!(target: LOG_TARGET, "Submitting liquidation extrinsic {opaque_tx:?}");
+										let _ = tx_pool_cc.submit_one(current_block_hash, TransactionSource::Local, opaque_tx.into()).await;
+										tracing::debug!(target: LOG_TARGET, "Liquidation execution time: {:?}", now.elapsed().as_millis());
+									}
+								);
+							}
+						}
+					}
+				});
+			}
+		}).await;
+	}
+
+	// TODO: return Result type
+	/// Fetch the preprocessed data used to evaluate possible candidates for liquidation.
+	async fn fetch_borrowers_data(http_client: HttpClient) -> Option<BorrowerData<AccountId>> {
+		let url = ("https://omniwatch.play.hydration.cloud/api/borrowers/by-health")
+			.parse()
+			.ok()?;
+		let res = http_client.get(url).await.ok()?;
+		assert_eq!(res.status(), StatusCode::OK);
```

Comment:
are we fine if this panics since we are in the node context?!

Reply:
fixed